command/ladder.js

```diff
@@ -29,16 +29,25 @@ function run(discordName, discordID, message, args, onComplete) {
             return;
         }
         var userMap = {};
-        var batchPayload = [];
+        var payload = [];
         for (var user of users) {
             userMap[user['steamId']] = user;
-            batchPayload.push({"platformId": user.platform, "uniqueId": user.steamId});
+            payload.push({"platformId": user.platform, "uniqueId": user.steamId});
+        }
+        var batchPayload = [];
+        for (var i = 0; i < payload.length; i += batchSize) {
+            batchPayload.push(payload.slice(i, i + batchSize));
         }
 
         // Next step, query for all their ranks in one go:
-        logger.info("Getting all stats for " + batchPayload.length + " users...");
-        rlClient.getStatsBatch(batchPayload).then(
-            function (userRatings) {
+        logger.info("Getting all stats for " + payload.length + " users...");
+        var statsPromises = [];
+        for (var batch of batchPayload) {
+            statsPromises.push(rlClient.getStatsBatch(batch));
+        }
+        Promise.all(statsPromises).then(
```

Comment:
sweet! I hadn't realized promises had an all(), this is a nice way to do it.

Reply:
Yea! it's pretty neat!